### DISCUSSION TITLE
Cask formula with multiple versions for multiple architectures

### DISCUSSION THREAD

**Discussion by stolendata** [Homebrew/discussions#5648] [Casks] [answered]
### Output of `brew config`

```shell
-
```

### Output of `brew doctor`

```shell
-
```

### Description of issue

I've just gotten around to finally release an Arm version of my mpv build. I now provide both "Intel" and an Arm builds but with different minimum macOS targets, and I also keep around legacy builds for past versions of OS X. I'd like to revise the cask formula to give Apple Silicon users access to the Arm build so they don't have to burden themselves with Rosetta and the x86-64 build, but I have too little time to dig into the documentation and figure out how to properly test a formula under different scenarios. Any quick suggestions or pointers?

### Relevant casks

stolendata-mpv

**Comment by gromgit** [comment, marked as answer]
https://docs.brew.sh/Cask-Cookbook#handling-different-system-configurations

**Comment by daeho-ro** [comment]
Before that, the arm version of cask should be signed well but it does not.

```
macOS on ARM requires software to be signed.
Please contact the upstream developer to let them know they should sign and notarize their software.
stolendata-mpv
```

**Reply by stolendata** [reply]
> Before that, the arm version of cask should be signed well but it does not.
> 
> ```
> macOS on ARM requires software to be signed.
> Please contact the upstream developer to let them know they should sign and notarize their software.
> stolendata-mpv
> ```

The app bundle is self-signed, which is sufficient, though it requires manual confirmation on first launch.

```
mba:~ sd$ codesign --verify --deep --verbose mpv-arm64-0.39.0/mpv.app
mpv-arm64-0.39.0/mpv.app: valid on disk
mpv-arm64-0.39.0/mpv.app: satisfies its Designated Requirement
```
